adoc/chapters/programming_interface.adoc

```diff
@@ -716,6 +716,40 @@ Construct a SYCL [code]#property_list# with zero or more properties.
 
 |====
 
+=== Information queries
+
+Each of the following classes provides a generic mechanism for querying the
+class for information: [code]#context#, [code]#device#, [code]#event#,
+[code]#kernel#, [code]#platform#, and [code]#queue#.
+
+Each available query is described by an <<information-descriptor>>, which is is
+a class or class template implementing the interface below:
+
+[source]
+----
+class __InformationDescriptor__
+{
+ public:
```

Comment:
I think the other part of the spec indent has `2` space, but that ok. Need to make a PR to format all the [source] with clang format